client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/ReplaceIfUnmodifiedOperation.java

```diff
@@ -35,6 +36,7 @@ protected VersionedOperationResponse executeOperation(Transport transport) {
       transport.writeArray(key);
       transport.writeVInt(lifespan);
       transport.writeVInt(maxIdle);
+      codec.writeExpirationNanoTimes(transport, lifespanNanos, maxIdleNanos, internalFlags);
```

Comment:
I think it's more neater to not write extra lifespan/maxIdle parameters, but instead to write them where you write lifespan/maxIdle already and just add the possibility to send a new option which defines the granularity of lifespan/maxIdle. 

Your current solution feels akward since you're adding two new fields and the existing ones are not used.

IOW, just add one more flag to the `flags` parameter here and you're done on the client :)